### PR TITLE
Fetch specific "id" task.

### PR DESCRIPTION
**Pull request by akib-raihan-sazim**
**Description of changes**
- Fetch task with specified taskID.

**Associated ticket link**
https://trello.com/c/K0dm2eXB

**Screenshot / Screen Recording**

https://github.com/user-attachments/assets/2fea4bfb-5204-4086-b235-4c92dd2df665

### CODE REVIEW

File: src/tasks/tasks.repository.ts [hunk 1]
```diff
@@ -32,4 +32,15 @@ export class TasksRepository extends EntityRepository<Task> {
...
+  async findOneTask(id: number): Promise<Task | null> {
+    try {
+      return await this.findOne({ id });
+    } catch (error) {
+      throw new InternalServerErrorException(
+        `Failed to find task with id ${id}`,
+        error.message,
+      );
+    }
+  }
```
Comment (ishmam-iqbal-sazim) [inline src/tasks/tasks.repository.ts:45]: Don't need this method. Since there are builtin repository methods, you can handle this in the service layer.

File: src/tasks/tasks.service.ts [hunk 2]
```diff
@@ -29,4 +33,22 @@ export class TasksService {
       );
     }
   }
+
+  async findTaskById(id: number): Promise<Task | null> {
+    try {
+      const task = await this.tasksRepository.findOneTask(id);
```
Comment (ishmam-iqbal-sazim) [inline src/tasks/tasks.service.ts:39]:
```suggestion
      const task = await this.tasksRepository.findOneOrFail(id);
```

Can use this builtin repository method. Not Found error exception need not be handled explicitly.

File: src/tasks/tasks.service.ts [hunk 3]
```diff
@@ -29,4 +33,22 @@ export class TasksService {
...
+      return task;
+    } catch (error) {
+      if (error instanceof NotFoundException) {
+        throw error;
+      }
+      throw new InternalServerErrorException(
+        `Failed to find task with ID ${id}`,
+        error.message,
+      );
+    }
```
Comment (ishmam-iqbal-sazim) [inline src/tasks/tasks.service.ts:52]: We don't explicitly need to throw internal server errors. Can remove this and the try catch block

File: src/tasks/tasks.controller.ts [hunk 4]
```diff
@@ -37,4 +39,19 @@ export class TasksController {
...
+    } catch (error) {
+      if (error instanceof NotFoundException) {
+        throw error;
+      }
+      throw new InternalServerErrorException(
+        `Failed to find task with ID ${id}`,
+        error.message,
+      );
+    }
+  }
```
Comment (ishmam-iqbal-sazim) [inline src/tasks/tasks.controller.ts:56]:
```suggestion
      return this.tasksService.findTaskById(Id)
```

Should be fine since you're handling the errors in service layer and there is no need to throw Internal Server Errors explicitly

File: src/tasks/tasks.service.ts [hunk 5]
```diff
@@ -22,4 +22,8 @@ export class TasksService {
   async findAllTasks(): Promise<Task[]> {
     return await this.tasksRepository.findAll();
   }
+
+  async findTaskById(id: number): Promise<Task | null> {
```
Comment (ishmam-iqbal-sazim) [inline src/tasks/tasks.service.ts:26]:
```suggestion
  async findOne(id: number): Promise<Task | null> {
```

Can name this findOne. It's not necessary to explicitly put tasks in the service method names since you'd be calling this method in other files with tasksService like so: `this.tasksService.findOne`

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left a comment

File: src/db/migrations/Migration20240802091219.ts [hunk 6]
```diff
@@ -1,13 +1,13 @@
-import { Migration } from '@mikro-orm/migrations';
+import { Migration } from "@mikro-orm/migrations";
 
 export class Migration20240802091219 extends Migration {
-
   async up(): Promise<void> {
-    this.addSql('create table "task" ("id" serial primary key, "title" varchar(255) not null, "summary" varchar(255) not null, "due_date" date null, "priority" text check ("priority" in (\'High\', \'Medium\', \'Low\')) not null, "completed" boolean not null default false);');
+    this.addSql(
```
Comment (ragib-hossain-sazim) [inline src/db/migrations/Migration20240802091219.ts:5]: [JC] Why this change is here?

Comment (akib-raihan-sazim) [reply src/db/migrations/Migration20240802091219.ts:5]: This was due to prettier formatting, I was using CLI for formatting. Will fix that.
